synapse/tests/test_lib_lmdb.py

```diff
@@ -0,0 +1,240 @@
+                self.false(setr.set(buid0, b'foo:lol', b'defv', flags=s_lmdb.STOR_FLAG_DEFVAL))
+                self.false(setr.has(b'foo:lol', b'defv'))
+                self.true(setr.has(b'foo:lol', b'rofl'))
+                self.true(psto.has(xact, b'foo:lol', b'rofl'))
+
+                self.true(setr.set(buid0, b'foo:lol', b'woo'))
+                self.true(setr.has(b'foo:lol', b'woo'))
+                self.true(psto.has(xact, b'foo:lol', b'woo'))
+                self.false(setr.has(b'foo:lol', b'defv'))
+                self.false(setr.has(b'foo:lol', b'rofl'))
+                self.false(psto.has(xact, b'foo:lol', b'defv'))
+                self.false(psto.has(xact, b'foo:lol', b'rofl'))
+
+                self.true(setr.set(buid1, b'foo:multi', b'root', flags=s_lmdb.STOR_FLAG_MULTIVAL))
+                self.true(setr.set(buid1, b'foo:multi', b'user', flags=s_lmdb.STOR_FLAG_MULTIVAL))
+                self.true(setr.has(b'foo:multi', b'root'))
+                self.true(setr.has(b'foo:multi', b'user'))
+                self.false(setr.set(buid1, b'foo:multi', b'user', flags=s_lmdb.STOR_FLAG_MULTIVAL))
+
+                retn = tuple(sorted(psto.pref(xact, b'e', b'gg')))
+                self.eq(retn, ())
+                retn = tuple(sorted(psto.pref(xact, b'f', b'gg')))
+                self.eq(retn, ())
+                retn = tuple(sorted(psto.pref(xact, b'foo:ba', b'asdf')))
+                self.eq(retn, ())
+                retn = tuple(sorted(psto.pref(xact, b'foo:baa', b'asdf')))
+                self.eq(retn, ())
+                retn = tuple(sorted(psto.pref(xact, b'foo:bar', b'asdf')))
+                self.eq(retn, ((buid0, b'foo:bar', b'asdfqwer'), (buid1, b'foo:bar', b'asdfzxcv')))
+                retn = tuple(sorted(psto.pref(xact, b'g', b'gg')))
+                self.eq(retn, ())
+
+                retn = tuple(sorted(psto.range(xact, b'foo:intish', b'\x00\x00', b'\x00\x80')))
+                self.eq(retn, ((buid1, b'foo:intish', b'\x00\x01'),))
+                retn = tuple(sorted(psto.range(xact, b'foo:nothere', b'\x00\x00', b'\x00\x80')))
+                self.eq(retn, ())
+
+                retn = tuple(sorted(psto.eq(xact, b'foo:intish', b'\x00\x01')))
+                self.eq(retn, ((buid1, b'foo:intish', b'\x00\x01'), ))
+                retn = tuple(sorted(psto.eq(xact, b'foo:nothere', b'\x00\x01')))
+                self.eq(retn, ())
+
+            buid2 = b'\x02' * 32
+            buid3 = b'\x03' * 32
+            rows = (
+                (buid2, b'hehe:a', b'haha'),
+                (buid2, b'hehe:b', b'haha'),
+                (buid3, b'hehe:c', b'haha'),
+            )
+            with lenv.begin(write=True) as xact:
+                setr = psto.getPropSetr(xact=xact)
+                setr.set(*rows[0])
+                setr.set(*rows[1])
+                # don't add the last row
+
+                retn = tuple(sorted(psto.recs(xact, rows)))
```

Comment:
@invisig0th @vEpiphyte - is this how recs is supposed to work?

Reply:
bump